src/ahrens_lab_to_nwb/yu_mu_cell_2019/yu_mu_cell_2019_imaging_extractor.py

```diff
@@ -0,0 +1,59 @@
+"""Modification of the roiextractors.Hdf5ImagingExtractor to read the custom miroscope data from the Ahrens lab."""
+from pathlib import Path
+from typing import Optional, Tuple
+
+import numpy as np
+from roiextractors.imagingextractor import ImagingExtractor
+from roiextractors.extraction_tools import PathType
+from lazy_ops import DatasetView
+
+
+try:
+    import h5py
+
+    HAVE_H5PY = True
+except ImportError:
+    HAVE_H5PY = False
+
+
+class AhrensHdf5ImagingExtractor(ImagingExtractor):
+    """Custom extractor for reading a single frame file from the Ahrens lab volumentric imaging data."""
+
+    extractor_name = "AhrensHdf5Imaging"
+    installed = HAVE_H5PY
+    mode = "file"
+    installation_mesg = "To use the Hdf5 Extractor run:\n\n pip install h5py\n\n"
+
+    def __init__(self, file_path: PathType, sampling_frequency: float):
+        ImagingExtractor.__init__(self)
+        self._kwargs = dict(file_path=str(Path(file_path).absolute()), sampling_frequency=sampling_frequency)
+        self._sampling_frequency = sampling_frequency
+        self.file_path = file_path
+
+        with h5py.File(name=file_path) as file:
+            self._num_stacks, self._num_rows, self._num_cols = file["default"].shape
+            self._dtype = file["default"].dtype
+        self._frame_axis_order = [1, 2, 0]
+
+    def get_video(self, start_frame: Optional[int] = None, end_frame: Optional[int] = None) -> np.ndarray:  # noqa: D102
+        with h5py.File(name=self.file_path) as file:
+            video = DatasetView(file["default"]).lazy_transpose(self._frame_axis_order)
+            return video.lazy_slice[:, :, :].dsetread()[np.newaxis, :]
+
+    def get_image_size(self) -> Tuple[int, int, int]:  # noqa: D102
+        return (self._num_rows, self._num_cols, self._num_stacks)
```

Comment:
`self.num_stacks`? I guess that's a good name for a tower of images. Did you see this somewhere?

Reply:
Not sure where I first saw it or why I even chose that name at the time. They aren't separate planes so the nomenclature of stuff like the SbxImagingExtractor isn't exact...

https://nwb-schema.readthedocs.io/en/latest/format.html?highlight=stack#sec-correctedimagestack-src is the closest reference to that I found on NWB.

Maybe the word just came up in reading the paper and talking with the experimenter